src/vec_f128_ppc.h

```diff
@@ -20,457 +20,930 @@
+ *  |power9   | 2     | 4/cycle  |
  *
- *	@param f128 a __float128 value containing a signed value.
- *	@return a __float128 value with magnitude from f128 and a positive
- *	sign of f128.
+ *  @param f128 a __float128 value containing a signed value.
+ *  @return a __float128 value with magnitude from f128 and a positive
+ *  sign of f128.
  */
 static inline __binary128
-__fabsf128 (__binary128 f128)
+vec_absf128 (__binary128 f128)
 {
-	vui32_t tmp;
-	const vui32_t signmask  = CONST_VINT128_W(0x80000000, 0, 0, 0);
-	__binary128 result;
-
+  __binary128 result;
 #if _ARCH_PWR9
-    __asm__(
-        "xsabsqp %0,%1;\n"
-        : "=v" (result)
-        : "v" (f128)
-        :);
+  __asm__(
+      "xsabsqp %0,%1;\n"
+      : "=v" (result)
+      : "v" (f128)
+      :);
 #else
-	tmp = vec_xfer_bin128_2_vui32t (f128);
-	tmp = vec_andc (tmp, signmask);
-	result = vec_xfer_vui32t_2_bin128 (tmp);
+  vui32_t tmp;
+  const vui32_t signmask = CONST_VINT128_W(0x80000000, 0, 0, 0);
+  tmp = vec_xfer_bin128_2_vui32t (f128);
+  tmp = vec_andc (tmp, signmask);
+  result = vec_xfer_vui32t_2_bin128 (tmp);
 #endif
-	return (result);
+  return (result);
 }
 
-/** \brief Copy the sign bit from f128y merge with magnitude from f128x
- * and return the resulting __float128 value.
+/** \brief Return true if the __float128 value is Finite
+ *  (Not NaN nor Inf).
  *
- *	@param f128x a __float128 value containing the magnitude.
- *	@param f128y a __float128 value containing the sign bit.
- *	@return a __float128 value with magnitude from f128x and the
- *	sign of f128y.
+ *  A IEEE Binary128 finite value has an exponent between 0x0000 and
+ *  0x7ffe (a 0x7fff indicates NaN or Inf).  The significand can be
+ *  any value.
+ *  Using the !vec_all_eq compare conditional verify this condition and
+ *  avoids a vector -> GPR transfer for platforms before PowerISA-2.07.
+ *  The sign bit is ignored.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   | 4-20  | 2/cycle  |
+ *  |power9   | 3     | 2/cycle  |
+ *
+ *  \note This function will not raise VXSNAN or VXVC (FE_INVALID)
+ *  exceptions. A normal __float128 compare can.
+ *
+ *  @param f128 a __float128 value in vector.
+ *  @return an int containing 0 or 1.
  */
-static inline __binary128
-__copysignf128 (__binary128 f128x , __binary128 f128y)
+static inline int
+vec_all_isfinitef128 (__binary128 f128)
 {
-	const vui32_t signmask  = CONST_VINT128_W(0x80000000, 0, 0, 0);
-	__binary128 result;
-	vui32_t tmpx, tmpy, tmp;
-
-#if _ARCH_PWR9
-    __asm__(
-        "xscpsgnqp %0,%2,%1;\n"
-        : "=v" (result)
-        : "v" (f128x), "v" (f128y)
-        :);
+#if defined (_ARCH_PWR9) && defined (scalar_test_data_class)
+  return !scalar_test_data_class (f128, 0x70);
 #else
-	tmpx = vec_xfer_bin128_2_vui32t (f128x);
-	tmpy = vec_xfer_bin128_2_vui32t (f128y);
+  vui32_t tmp, t128;
+  const vui32_t expmask = CONST_VINT128_W(0x7fff0000, 0, 0, 0);
+  int result = 0;
```

Comment:
don't need to initialize `result`.  really don't need `result` at all if final assignment changed to `return !vec_all_eq...`